### ISSUE TITLE
Helpers calls being stacked every time a route is re-visited

### ISSUE THREAD

**Issue by michaelunomena** [meteorhacks/flow-layout#33]
I have a structure as follows:

template name="mainLayout":
---famousHeader
------Template.dynamic name="header"
---famousContent
------Template.dynamic name="main"
---famousFooter
------Template.dynamic name="footer"

----------------------------------

Between the following 2 routes, I then render as follows:

/ -> FlowLayout.render("mainLayout", {header: "mainHeader", main: "home", footer: "mainFooter"});

/another-route -> FlowLayout.render("mainLayout", {header: "mainHeader", main: "anotherRoute", footer: "mainFooter"});

-----------------------------------

/ and /another-route both use helpers which I put a console.log("calling home") and console.log("calling another route") in respectively.

When I hit /, "calling home" prints once, when I hit /another-route, "calling another route" prints once, however when I hit the browser back button or FlowRouter.go("/") from /another-route and the browser navigates back to / from /another-route, it then prints "calling home" twice, and every time I go to to /another-route and back to / it will print "calling home" 1 additional time, which points me towards thinking that these helpers are somehow being registered every time FlowLayout does an _updateRegions call (as I am using the same layout between the 2 routes).

I have spent days trying to debug this and learnt a lot about Meteor's internals by going through this process but I have yet to find the cause of the problem. If anyone has experienced something similar or can point me in the right direction as to why these helpers are being called multiple times it would be greatly appreciated!

**Comment by arunoda** [comment]
I did a related fix few hours ago and published a new version. Could you try that?
If not, I'd love to see repo.

**Comment by michaelunomena** [comment]
Hi arunoda

Thanks alot, I'll give it a try when I get home today otherwise I'll setup a test repo replicating this.

Kind Regards
Michael

**Comment by arunoda** [comment]
Great.

**Comment by michaelunomena** [comment]
Hi arunoda

I upgraded to FlowLayout 1.4.2 and that didn't help the problem so it seems the issue is unrelated.

It seems I have finally found the root of the problem however. If I remove my isReady "subscriptionName" helper to find out if the subscription is ready, my Each call still works because of data already cached in minimongo and the stacking of helper calls goes away between routes. The reactivity in the isReady helper call must be causing some funnies. I'll do some further debugging on this tomorrow.
